libcontainer/user/user.go

```diff
@@ -258,35 +255,47 @@ func GetExecUser(userSpec string, defaults *ExecUser, passwd, group io.Reader) (
 		user.Sgids = []int{}
 	}
 
-	// allow for userArg to have either "user" syntax, or optionally "user:group" syntax
+	// Allow for userArg to have either "user" syntax, or optionally "user:group" syntax
+	var userArg, groupArg string
 	parseLine(userSpec, &userArg, &groupArg)
 
+	// Find the matching user.
 	users, err := ParsePasswdFilter(passwd, func(u User) bool {
 		if userArg == "" {
+			// Default to current state of the user.
 			return u.Uid == user.Uid
 		}
-		return u.Name == userArg || strconv.Itoa(u.Uid) == userArg
+
+		if uid, err := strconv.Atoi(userArg); err == nil {
```

Comment:
Do you think it'd be worthwhile or possible to do this test/conversion once instead of once per `passwd` entry like it's doing now, plus one more time outside this filter if we don't find the UID in the file?